File: payment-service/src/main/java/com/hedvig/paymentservice/domain/payments/Transaction.java
package com.hedvig.paymentservice.domain.payments;

import lombok.AllArgsConstructor;
import lombok.Data;

import javax.money.MonetaryAmount;
import java.time.Instant;
import java.util.UUID;

@Data
@AllArgsConstructor
public class Transaction {
  UUID transactionId;

  MonetaryAmount amount;
  Instant timestamp;
  TransactionType transactionType;
  TransactionStatus transactionStatus;
}

File: payment-service/src/main/java/com/hedvig/paymentservice/domain/payments/Transaction.kt
package com.hedvig.paymentservice.domain.payments

import java.time.Instant
import java.util.UUID
import javax.money.MonetaryAmount

data class Transaction(
    val transactionId: UUID,
    val amount: MonetaryAmount,
    val timestamp: Instant
) {
  lateinit var transactionType: TransactionType
  lateinit var transactionStatus: TransactionStatus
}
